cypress/integration/admin_portal/helpers/enterprise_coupons.js

```diff
@@ -0,0 +1,87 @@
+const uuidv4 = require('uuid/v4')
+
+function getDates() {
+  let dates = {}
+  let target_date = new Date()
+  target_date.setDate(target_date.getDate() - 3)
+  dates.start_date = target_date.toISOString()
+  target_date.setDate(target_date.getDate() + 20)
+  dates.end_date = target_date.toISOString()
+  return dates
+}
+
+
+class EnterpriseCoupons {
+
+  loginToEcommerce() {
+    let token
+    cy.login_using_api(Cypress.env('ADMIN_PORTAL_USER_EMAIL'), Cypress.env('ADMIN_PORTAL_USER_PASSWORD'))
+    cy.request(`${Cypress.env('ecommerce_url')}/coupons/`)
+    cy.getCookie('ecommerce_csrftoken')
+      .should('exist').then((csrf_val) => {
+        this.token = csrf_val
+      })
+  }
+
+  prepareCouponData(coupon) {
+    cy.fixture('coupon_creation_data').as('couponRequestBody')
+    const randomString = uuidv4().substr(-11)
+    const couponTitle = `Test_Coupon_${randomString}`
+    return cy.get('@couponRequestBody').then((requestBody) => {
+      requestBody[coupon]['title'] = couponTitle
+      requestBody[coupon]['enterprise_customer']['id'] = Cypress.env('enterprise_id')
+      requestBody[coupon]['enterprise_customer']['name'] = Cypress.env('enterprise_name')
+      requestBody[coupon]['enterprise_customer_catalog'] = Cypress.env('enterprise_catalog')
+      requestBody[coupon]['start_datetime'] = getDates().start_date
+      requestBody[coupon]['end_datetime'] = getDates().end_date
+    })
+  }
+
+  createCoupon(requestBody) {
+    let createCouponUrl = `${Cypress.env('ecommerce_url')}/api/v2/enterprise/coupons/`
+    return cy.request({
+      method: 'POST',
+      url: createCouponUrl,
+      body: requestBody,
+      headers: {
```

Comment:
`headers` here and in `deleteCoupon` function are same, can we create a util function to return the headers with any dynamic data passed to that function to be substituted?